das_api/src/error.rs

```diff
@@ -0,0 +1,24 @@
+use {jsonrpsee::core::Error as RpcError, jsonrpsee::types::error::CallError, thiserror::Error};
+
+#[derive(Error, Debug)]
+pub enum DasApiError {
+    #[error("Config Missing or Error {0}")]
+    ConfigurationError(String),
+    #[error("Server Failed to Start")]
+    ServerStartError(#[from] RpcError),
+    #[error("Database Connection Failed")]
+    DatabaseConnectionError(#[from] sqlx::Error),
+    #[error("Pubkey Validation Err {0} is invalid")]
+    PubkeyValidationError(String),
+    #[error("Validation Error {0}")]
+    ValidationError(String),
+    #[error("Database Error")]
+    DatabaseError(#[from] sea_orm::DbErr),
+}
+
+impl Into<RpcError> for DasApiError {
```

Comment:
If you implement the `From` trait instead of the `Into` trait, I believe errors can be converted implicitly **without** having to use  `map_err(Into::into)` as I see you are using in `api.rs`.  Basically `from()` will automatically be used to do the conversion.

```
impl From<DasApiError> for RpcError {
    fn from(error: DasApiError) -> Self
       ...
    }
}